infrastructure/src/main/java/org/corfudb/infrastructure/protocol/NettyServerRouter.java

```diff
@@ -60,6 +62,17 @@ public NettyServerRouter(ImmutableList<AbstractServer> servers, ServerContext se
         return Optional.ofNullable(serverContext.getCurrentLayout());
     }
 
+
+
+    @Override
+    public void sendResponse(Response response, ChannelHandlerContext ctx) {
+        ByteBuf outBuf = PooledByteBufAllocator.DEFAULT.buffer();
+        outBuf.writeByte(API.PROTO_CORFU_MSG_MARK);
+        outBuf.writeBytes(response.toByteArray());
+        //TODO(Zach): Where to remove allocation?
+        ctx.writeAndFlush(outBuf);
+    }
```

Comment:
Pass the allocator via the constructor. Accessing a resources from multiple entry points makes it hard to change. You can wrap the default allocator and pass that instance to the constructor.